### ISSUE TITLE
Non-ascii comments are not displayed correctly

### ISSUE THREAD

**Issue by hakavlad** [Picocrypt/Picocrypt#23]
Non-ascii comments are displayed as `?????????????`. Can this be fixed?

**Comment by HACKERALERT** [comment]
Not too sure, I believe it might be because the font doesn't support it. I forget what the font was exactly because I named it font.ttf 🤦 but it might be Noto Sans. Not a big deal, 3 possible solutions:
1. Change font. Ideally not, some of the UI has been designed around the particular font size/width
2. Add note to README. This is easy and should be done.
3. Don't allow users to type non-ascii characters into the comments box.

**Comment by hakavlad** [comment]
>might be because the font doesn't support it.

No. For example, all popular fonts suppots basic non-ascii symbols like ё and —.

The problem occurs on both Windows and Linux.

>but it might be Noto Sans

The problem is not that exotic characters are not supported. The most basic things are not supported!

I suppose it's possible to do something with the input encodings.

>3 possible solutions

1. Not works.
2. Not a solution.
3. Not a solution.

**Comment by HACKERALERT** [comment]
What do you think could fix it?

**Comment by hakavlad** [comment]
I'm not gopher. Maybe consult on SO or r/golang.

**Comment by HACKERALERT** [comment]
Go itself supports Unicode character in strings, I think it's just the UI library "dear imgui" that is having trouble.

**Comment by hakavlad** [comment]
Maybe related

>unspecified font range defaults to minimal latin range, it does not include all characters provided by font.

https://github.com/ocornut/imgui/issues/2233

**Comment by hakavlad** [comment]
see https://pkg.go.dev/github.com/allendang/imgui-go#Font and further

**Comment by hakavlad** [comment]
see also https://github.com/inkyblackness/imgui-go/issues/123#issuecomment-692470183

**Comment by HACKERALERT** [comment]
Not a big priority imo

**Comment by Retengart** [comment]
There is a solution, but it is a little complicated. To use giu version 0.7.0, but it has to be optimized 
![image](https://github.com/user-attachments/assets/1bc8fd0d-8e58-4e60-a079-78fc7a0c8d63)

**Comment by Retengart** [comment]
In the example above, I just re-imported the version of giu 0.7.0 from the original author AllenDang

**Comment by HACKERALERT** [comment]
The problem with 0.7.0 and why I freezed giu at 0.5.7 is because high dpi screen scaling breaks in 0.7.0, as can been seen in your screenshot. Moving to 0.7.0 is possible, but will need a ton of testing to ensure existing behaviour does not break. Don't think question marks are a big enough reason to make those potentially dangerous changes. Feel free to try it out and put it in the Experiments repo in case someone wants to revisit it in the future.

**Comment by Retengart** [comment]
Yeah, without nasty tests it's difficult to say something. But with DPI we can fight. For example, this is the giu 0.6.2 and there are no problems with scaling. But the application may crash accidentally
![image](https://github.com/user-attachments/assets/999cd167-b149-4852-b2fa-eb5c7180c80c)

**Comment by Retengart** [comment]
And also it's impossible to use ClearActivityID() with new giu
![image](https://github.com/user-attachments/assets/9699585d-78ff-4434-83fb-9c876afedaaf)

**Comment by HACKERALERT** [comment]
The thing about scaling is that looking at one display is not enough, you need to drag it to a screen with a different pixel density for potential issues to show. It may work fine on a laptop screen but scale poorly on a large monitor. Like you mentioned, newer versions have a bunch of issues like random crashing. There used to be crashes in the currently used version as well and it took a while to fix and it was a fix in C. Too much work.

If we're gonna spend time on ui updates, better to switch to a more future proof UI library like gioui.org.

**Comment by Retengart** [comment]
How about adding a new additional clause with symbols for generating a password?
![image](https://github.com/user-attachments/assets/dcd24455-4345-40f3-9bdc-1a65c1bedcdf)
For example this:
`if passgenExtendedASCII {
		chars += "~`%*.,/|\\[]{}:;"
	}`
By default, this feature is disabled in the "Create Password" window

**Comment by HACKERALERT** [comment]
If it's just a few characters added, probably not that important? It might break other password managers or file formats, so best to stay safe with standard letters. You can always increase the length.

**Comment by HACKERALERT** [comment]
Closing as part of a cleanup effort.
